### PR TITLE
Feature/reset password fix

### PR DESCRIPTION
**Pull request by WinnMoo**
# Issue/Ticket

PROGRESS ON #XX

If you specify CLOSES, the ticket will be automatically closed and moved to 'done' when the PR is merged.

If you specify PROGRESS ON, a reference will be created to that issue (in both directions), but merging the PR will not close the issue.

# Description

Refactored user controller/reset controller. Refactored reset password UI to be less clunky. Added update password logic/ui, still needs to be tested manually. 

## Type of change

Feature

Label your PR accordingly.

# How Has This Been Tested?

Please describe and list the tests that you wrote for each module changed. If there are any special considerations, mention them here.

- Manual testing for reset password

### CODE REVIEW

File: Backend/DataAccessLayer/DataAccessLayer.csproj
Comment (alfredovargasjr) [inline Backend/DataAccessLayer/DataAccessLayer.csproj]: Have you tried running these migrations on your local db?

File: Backend/ManagerLayer/Login/LoginManager.cs
Comment (alfredovargasjr) [inline Backend/ManagerLayer/Login/LoginManager.cs]: Why is this commented out?

File: Backend/ManagerLayer/AuthorizationManagement/AuthorizationManager.cs
Comment (alfredovargasjr) [inline Backend/ManagerLayer/AuthorizationManagement/AuthorizationManager.cs]: Merge conflict here

File: Backend/ServiceLayer/Services/EmailService.cs
Comment (alfredovargasjr) [inline Backend/ServiceLayer/Services/EmailService.cs:27]: Reason for this?

Comment (alfredovargasjr) [reply Backend/ServiceLayer/Services/EmailService.cs:27]: Can you make it like 'noreply-support@kfcsso.com'?

File: Frontend/src/views/Login.vue
Comment (alfredovargasjr) [inline Frontend/src/views/Login.vue]: Merge conflicts

**Review by alfredovargasjr — CHANGES_REQUESTED** [review]

File: Backend/ManagerLayer/Login/LoginManager.cs
Comment (WinnMoo) [reply Backend/ManagerLayer/Login/LoginManager.cs]: Not too sure, I pulled that from login-frontend-updated

File: Backend/ServiceLayer/Services/EmailService.cs
Comment (WinnMoo) [reply Backend/ServiceLayer/Services/EmailService.cs:27]: SES for the website will need to be setup for it to work

File: Backend/ManagerLayer/Login/LoginManager.cs
Comment (julianpoy) [reply Backend/ManagerLayer/Login/LoginManager.cs]: Probably not the best idea to merge other incomplete branches into your branch. Now you're dependent on those changes being merged into master before this PR can be approved.

File: Backend/DataAccessLayer/App.config
Comment (julianpoy) [inline Backend/DataAccessLayer/App.config]: This file should not be changed.

File: Backend/DataAccessLayer/Migrations/201903122358472_pending-migration.cs
Comment (julianpoy) [inline Backend/DataAccessLayer/Migrations/201903122358472_pending-migration.cs]: The reason this "fake" migration is needed is because you modified an existing migration.

It's really best to add a new migration that updates that table.

If you can't find a way (as we discussed), the best approach is probably to remove these changes and just create a new migration that drops the existing table, and creates a new table with the correct schema.

File: Backend/Dev/App.config
Comment (julianpoy) [inline Backend/Dev/App.config]: These changes shouldn't be necessary - please revert unless you have a specific reason for adding these.

File: Backend/KFC_WebAPI/Controllers/UsersController.cs
Comment (julianpoy) [inline Backend/KFC_WebAPI/Controllers/UsersController.cs]: This PR is for password reset - why is it modifying login components?

File: Backend/KFC_WebAPI/KFC_WebAPI.csproj
Comment (julianpoy) [inline Backend/KFC_WebAPI/KFC_WebAPI.csproj]: This file should not be changed.

File: Backend/KFC_WebAPI/Web.config
Comment (julianpoy) [inline Backend/KFC_WebAPI/Web.config]: This file should not be changed.

File: Backend/ManagerLayer/packages.config
Comment (julianpoy) [inline Backend/ManagerLayer/packages.config]: This file should not be changed.

File: Backend/UnitTesting/LoginManagerUT.cs
Comment (julianpoy) [inline Backend/UnitTesting/LoginManagerUT.cs]: Again, modifying login code here. I believe this is from your merge. You'll have to wait until the login PR is merged before this PR is merged. Undoubtedly there will be changes you have to revert as well, since his PR has changed since you merged that branch into this one.

File: Frontend/package.json
Comment (julianpoy) [inline Frontend/package.json:20]: What do we need vuex for?

File: Frontend/src/store/modules/User.js
Comment (julianpoy) [inline Frontend/src/store/modules/User.js]: [non-blocking] Hmm, not sure that having a VueX store is really necessary for functionality. Not my feature though, so I'll let it pass.

File: Frontend/src/views/Register.vue
Comment (julianpoy) [inline Frontend/src/views/Register.vue]: This file should not be changed.

**Review by julianpoy — CHANGES_REQUESTED** [review]
See comments

**Pull request closed without merging**
